### ISSUE TITLE
Bug: [PathValue] <PathValue returns undefined for object paths>

### ISSUE THREAD

**Issue by vmuckxyz** [ts-essentials/ts-essentials#406]
### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `typescript`                           | `5.5.4` |
| `ts-essentials`                        | `10.0.1` |

### Repro Code

```typescript
import {Paths, PathValue} from 'ts-essentials';

interface Company {
  name: string;
  employees: { name: string }[];
}

type CompanyPaths = PathValue<Company, `employees.0`>;
//   ^?

function get<Type, StringPath extends string>(obj: Type, path: `${StringPath}`): PathValue<Type, StringPath>;
function get<Type>(obj: Type, path: string) {
  const keys = (path as string).split(".") as (keyof Type)[];

  return keys.reduce((acc, key) => acc?.[key] as unknown as Type, obj);
}

declare const company: Company;

const companyName = get(company, 'name');
//    ^?
const companyEmployees = get(company, 'employees');
//    ^?
const companyEmployee = get(company, 'employees.0');
//    ^?
const companyEmployeeName = get(company, 'employees.0.name');
//    ^?
```

### TypeScript Playground

https://tsplay.dev/m3qQjw

### Expected Result

I expect the types two-slashed in the example to match the README.md in the ts-essentials git repo: https://github.com/ts-essentials/ts-essentials/tree/master/lib/path-value

expected:
```
type CompanyPaths = PathValue<Company, `employees.0`>;
//   ^? { name: string } | undefined
...
const companyEmployeeName = get(company, "employees.0.name");
//    ^? const companyEmployeeName: undefined
```

### Actual Result

All types two-slashed in the example are showed to be undefined.

```
type CompanyPaths = PathValue<Company, `employees.0`>;
//   ^? type CompanyPaths = undefined
...
const companyEmployeeName = get(company, "employees.0.name");
//    ^? const companyEmployeeName: undefined
```

**Comment by vmuckxyz** [comment]
And since I have a nasty habit of forgetting some information after I click "submit", I am able to reproduce locally as well in a Vite application using the above versions of TypeScript and ts-essentials.

**Comment by Beraliv** [comment]
@vmuckxyz if you'd like to have a quick fix, please replace `interface` with `type` - https://tsplay.dev/NnY9qw

At the same time, I'm applying a fix to interface at any level, I will prepare a fix shortly

**Comment by vmuckxyz** [comment]
I can confirm that moving from interface to type works as expected on my local environment as well.

**Comment by Beraliv** [comment]
@vmuckxyz I've also prepared a fix in https://github.com/ts-essentials/ts-essentials/pull/407, I'm sure that the fix is effective, but could you please run it through your project in case I'm missing something?

**Comment by vmuckxyz** [comment]
I can confirm both in tsplay and in my local repo that this fixes not only interfaces, but also union types that include interfaces (e.g. `type foo = Interface1 & Interface2`).

**Comment by Beraliv** [comment]
@vmuckxyz great, thank you for double-checking the fix. I've merged my PR and will deploy the patch version (`ts-essentials@10.0.2`) shortly

**Comment by Beraliv** [comment]
The fix is available in `ts-essentials@10.0.2` - https://github.com/ts-essentials/ts-essentials/releases/tag/v10.0.2. Thank you for your help @vmuckxyz!
